docs/02-app/02-api-reference/04-functions/use-pathname.mdx

```diff
@@ -36,6 +36,7 @@ For example, a Client Component with `usePathname` will be rendered into HTML on
 > - Reading the current URL from a [Server Component](/docs/app/building-your-application/rendering/server-components) is not supported. This design is intentional to support layout state being preserved across page navigations.
 > - Compatibility mode:
 >   - `usePathname` can return `null` when a [fallback route](/docs/pages/api-reference/functions/get-static-paths#fallback-true) is being rendered or when a `pages` directory page has been [automatically statically optimized](/docs/pages/building-your-application/rendering/automatic-static-optimization) by Next.js and the router is not ready.
+>   - When using `usePathname` with [rewrites](/docs/app/api-reference/next-config-js/rewrites), `useState` and `useEffect` must also be used in order to avoid hydration mismatch errors. See the [rewrites example](https://github.com/vercel/next.js/tree/canary/examples/rewrites) for more information.
```

Comment:
```suggestion
>   - When using `usePathname` with [rewrites](/docs/app/api-reference/next-config-js/rewrites), `useState` and `useEffect` must also be used in order to avoid hydration mismatch errors. See the [rewrites example](https://github.com/vercel/next.js/tree/canary/examples/rewrites) for more information.
```

This also applies to middleware rewrites